specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/EasyAuthConfigs.json

```diff
@@ -0,0 +1,706 @@
+    },
+    "ClientRegistration": {
+      "description": "The configuration settings of the app registration for providers that have client ids and client secrets",
+      "type": "object",
+      "properties": {
+        "clientId": {
+          "description": "The Client ID of the app used for login.",
+          "type": "string"
+        },
+        "clientSecretRefName": {
+          "description": "The app secret ref name that contains the client secret.",
+          "type": "string"
+        }
+      }
+    },
+    "Google": {
+      "description": "The configuration settings of the Google provider.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "<code>false</code> if the Google provider should not be enabled despite the set registration; otherwise, <code>true</code>.",
+          "type": "boolean"
+        },
+        "registration": {
+          "$ref": "#/definitions/ClientRegistration",
+          "description": "The configuration settings of the app registration for the Google provider."
+        },
+        "login": {
+          "$ref": "#/definitions/LoginScopes",
+          "description": "The configuration settings of the login flow."
+        },
+        "validation": {
+          "$ref": "#/definitions/AllowedAudiencesValidation",
+          "description": "The configuration settings of the Azure Active Directory token validation flow."
+        }
+      }
+    },
+    "AllowedAudiencesValidation": {
+      "description": "The configuration settings of the Allowed Audiences validation flow.",
+      "type": "object",
+      "properties": {
+        "allowedAudiences": {
+          "description": "The configuration settings of the allowed list of audiences from which to validate the JWT token.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "Twitter": {
+      "description": "The configuration settings of the Twitter provider.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "<code>false</code> if the Twitter provider should not be enabled despite the set registration; otherwise, <code>true</code>.",
+          "type": "boolean"
+        },
+        "registration": {
+          "$ref": "#/definitions/TwitterRegistration",
+          "description": "The configuration settings of the app registration for the Twitter provider."
+        }
+      }
+    },
+    "TwitterRegistration": {
+      "description": "The configuration settings of the app registration for the Twitter provider.",
+      "type": "object",
+      "properties": {
+        "consumerKey": {
+          "description": "The OAuth 1.0a consumer key of the Twitter application used for sign-in.\nThis setting is required for enabling Twitter Sign-In.\nTwitter Sign-In documentation: https://dev.twitter.com/web/sign-in",
+          "type": "string"
+        },
+        "consumerSecretRefName": {
```

Comment:
>consumerSecretRefName [](http://example.com/codeflow?start=9&length=21)

Add x-ms-secret extension if this is a secret.

Reply:
Not secret as above